lib/web_ui/lib/src/engine/text/layout_service.dart

```diff
@@ -263,6 +263,57 @@ class TextLayoutService {
     }
     return boxes;
   }
+
+  ui.TextPosition getPositionForOffset(ui.Offset offset) {
```

Comment:
Add the comment: // After layout, each paragraph span should have info about
    // its position and dimensions. Use that information to find which span the
    // offset belongs to, then search within that span for the exact character.

Reply:
Done.